doc/spec_dataValidityPropagation.dox

```diff
@@ -0,0 +1,90 @@
+- 2.3.4 Inside the ApplicationModule main loop the module's data fault counter is accessible. The user can increment and decrement it, but has to be careful to do this in pairs. The more common use case will be to query the module's data validity.
+
+### 2.4 TriggerFanOut
+\anchor dataValidity_2_4
+ 
+The TriggerFanOut is special in the sense that it does not compute anything, but reads multiple independent poll-type inputs when a trigger arrives, and pushes them out. In contrast to an ApplicationModule or one of the data fan-outs, their data validities are not connected.
+
+- 2.4.1 Only the push-type trigger input of the TriggerFanOut is equiped with a MetaDataPropagatingRegisterDecorator.
+- \anchor dataValidity_2_4_2 2.4.2 The poll-type data inputs do not have a MetaDataPropagatingRegisterDecorator. \ref dataValidity_comment_2_4_2 "(*)".
+
+- 2.4.3 The individual poll-type inputs propagate the data validity flag only to the corresponding outputs.
+- \anchor dataValidity_2_4_4 2.4.4 Although the trigger conceptually has data type 'void', it can also be `faulty` \ref dataValidity_comment_2_4_4 "(*)". An invalid trigger will result be processed, but all data that is being read out is flagged as `faulty`.
+
+### 2.5 Interaction with exception handling
+
+See @ref spec_execptionHandling.
+
+- 2.5.1 When creating the objects, ApplicationCore takes care that the MetaDataPropagatingRegisterDecorator is always placed around the ExceptionHandlingDecorator if both decorators are used. Like this a `faulty` flag raised by the ExceptionHandlingDecorator is automatically picked up by the MetaDataPropagatingRegisterDecorator.
+- 2.5.2 The first failing read returns with the old data and the 'faulty' flag. Like this the flag is propagated to the outputs. Only further reads might freeze until the device is available again.
+
+### 2.6 Application class
+
+- 2.6.1 For device variables, the requirement of setting receiving endpoints to 'faulty' on construction can not be fulfilled. In DeviceAccess the accessors are bidirectional and provide no possibility to distinguish sender and receiver. Instead, the validity is set right after construction in Application::createDeviceVariable() for receivers.  
+
+### Comments:
+
+- 2.1 The MetaDataPropagatingRegisterDecorator also propagates the version number, not only the data validity flag. Hence it's not called DataValidityPropagatingRegisterDecorator.
+- \anchor dataValidity_comment_2_3_3a \ref dataValidity_2_3_3 "2.3.3" If there would be some outputs which are still valid if a particular input is faulty it means that they are not connected to that input. This usualy is an indicator that the module is doing unrelated things and should be split.
```

Comment:
```suggestion
- \anchor dataValidity_comment_2_3_3a \ref dataValidity_2_3_3 "2.3.3" A Module having a valid output, when a particular input is faulty indicates that the input is not connected to that output. This is an indicator that the module is doing unrelated things and should be split.
```

Reply:
We need the subjunctive in the first part, because the module cannot have an output that is valid if an input is invalid.
```
If a module has an output that would still be valid, even though one of the inputs is faulty, it means that the output is not connected to that particular input. This is an indicator that the module is doing unrelated things and probably should be split.
```